src/components/Panels/Telescope/Targets/UpdateGuideTargets.tsx

```diff
@@ -22,12 +22,12 @@ export function UpdateGuideTargets({ canEdit }: { canEdit: boolean }) {
         observationId: configuration!.obsId!,
         observationTime: crtTime,
       },
-      onCompleted(data) {
+      async onCompleted(data) {
         const OiwfsTargets: TargetInput[] = [];
         const Pwfs1Targets: TargetInput[] = [];
         const Pwfs2Targets: TargetInput[] = [];
-        data.observation?.targetEnvironment.guideEnvironments.map((env) => {
-          env.guideTargets.map((t) => {
+        data.observation?.targetEnvironment.guideEnvironments.forEach((env) => {
+          env.guideTargets.forEach((t) => {
             const auxTarget: TargetInput = {
               name: t.name,
               id: undefined,
```

Comment:
Good, thank you for noticing that forEach was a better choice than map for this case.
By the way, with Todd work, we should change the query and the expected response is going to be only one guide target, then this array logic will change :cry:

Reply:
What work do you mean?

Reply:
Todd is working on a PR to add a new way to query the guide target which is the one we should use, that query will request the observation duration as input to decide which is the best guide target and will return only one option instead of a list like it is doing it currently.
Anyways, he mentioned that the current query navigate is using, won't be removed until the new one is completely integrated.